### PR TITLE
[TS-51] Follow List and Follow/Unfollow Action

### PR DESCRIPTION
**Pull request by Ming424**
Video: https://youtu.be/tCwbBwt8U6I
There are a lot of console.log and 1 button [State] inside otherdashboard, they are removed.

TS-51
image

+ Login state for follower/following
+ Middleware for follow related action
+ Live following/follower
+ Connect other dashboard with backend (Follow, Position)
+ Update btn style at linking 3rd party APP

New dependency: vuex-persistedstate

OLD:
This PR contains TS-327 and TS-397, move TS-328 to next sprint
image

Followings
image
Followers
image

Snackbar notification for the success response
image

When list is empty
image

Sidebar (fetch at login, dont know how to refresh, see Issues below)
image

New asset
image

Issue (I think we need a new ticket for these): 
- No profile picture implement, everything will be set to default (above asset)
- Slow when fetching the list
- Not sure how to refresh number in sidebar once follow/unfollow happened
- Remove dummy data, the tests are failing, I commented them out

Current relationship (API https://documenter.getpostman.com/view/10545271/UVe9QU3y#b55383f2-6f80-4271-be4f-a0514a28f0d6)
image

Accounts I use to test
test                 [6204a777f82a1fb5c9f80a1f]
TianMing        [6211cb412ab77bf00f22fd87]
TianMing2      [6211cb5a2ab77bf00f22fd91]
IDNF Pico       [6212d380693803381b1583a6]

HOW TO TEST

Add User ID in Line 41 in UserBlock to debug easier
image

Copy accessToken once login
image

Pasted inside Authorization-Bearer_Token

image

Try to follow and unfollow from [Following] list (same logic in [Followers] list, but [Following is easier to restore)
Once follow/unfollow, DO
GET http://localhost:5000/api/v1/following/follows/6204a777f82a1fb5c9f80a1f
To see the list of following match you expectation
image

If you accidently lost a relation, how to follow in postman
- From the same page without refreshing
- POST http://localhost:5000/api/v1/following/follow
- - {
    "actorId": "6204a777f82a1fb5c9f80a1f",
    "targetId": "{ID_YOU_WANT_TO_FOLLOW}"
}

Still fixing the E2E, but i wont affect the testing

### CODE REVIEW

**Comment by AlyaNaseer** [comment]
Wow this is a relatively big PR haha! 

A few things I noted:
image

1. There is too much spacing above and below the text, which is forcing a scroll, which in this case we don't need. You could look at just centering the text vertically so it doesn't make the component bigger
2. I think we should change the text from "Empty List" to "You aren't following anyone"/"You have no followers"
3. In the side menu, if there are no followers/following, it should say 0 instead of "null"
4. Did you remove the search in the following page altogether? Or only when you aren't following anyone?

As for the cypress tests, we'll need to make a decision about what these pages should look like and how they should behave so that we either keep the uncommented code or we remove (we have to make sure not to push it as is as it isn't good programming practices to have that much commented out code)

**Review by AlyaNaseer — CHANGES_REQUESTED** [review]
See changes mentioned above

**Comment by AlyaNaseer** [comment]
You've got a failing Cypress test as well 
image

File: src/pages/Followers/Followers.vue [hunk 1]
```diff
@@ -10,19 +11,49 @@
...
+          >
+            <v-container
+              fill-height
+              fluid
+            >
+              <v-row
+                align="center"
+                justify="center"
+              >
+                <v-col>You do not have any follower yet.</v-col>
```
Comment (itsShaVawn) [inline src/pages/Followers/Followers.vue:52]:
```suggestion
                <v-col>You do not have any followers yet.</v-col>
```

**Review by itsShaVawn — CHANGES_REQUESTED** [review]
The side menu doesnt appear for some reason? 
image

the following page looks like this?
image

**Comment by itsShaVawn** [comment]
You also have merge conflicts, code duplication and 2 code smells. The code smells are commented code, so just remove this.

**Comment by Ming424** [comment]
> The side menu doesnt appear for some reason? image
> 
> the following page looks like this? image

Because it auto login, sign off and login will solve the issue.
Following is fixed
image

File: src/pages/Followers/Followers.vue (hunk 1 above)
Comment (Ming424) [reply src/pages/Followers/Followers.vue:52]: Fixed

**Comment by Ming424** [comment]
> Wow this is a relatively big PR haha!
> 
> A few things I noted: image
> 
> 1. There is too much spacing above and below the text, which is forcing a scroll, which in this case we don't need. You could look at just centering the text vertically so it doesn't make the component bigger
> 2. I think we should change the text from "Empty List" to "You aren't following anyone"/"You have no followers"
> 3. In the side menu, if there are no followers/following, it should say 0 instead of "null"
> 4. Did you remove the search in the following page altogether? Or only when you aren't following anyone? Because I noticed you commented out a lot of cypress tests
> 
> As for the cypress tests, we'll need to make a decision about what these pages should look like and how they should behave so that we either keep the uncommented code or we remove (we have to make sure not to push it as is as it isn't good programming practices to have that much commented out code)

+ For num 0 
image
+ For empty list, was obsessed with vertical center, but have problem with, so it will be bind to top
image
+ Cypress is fixed

**Comment by AlyaNaseer** [comment]
Idk why but for some reason I'm having a hard time pulling your branch, it doesn't seem to be updated. But from the video, everything looks good to me

**Review by AlyaNaseer — APPROVED** [review]

**Comment by itsShaVawn** [comment]
I'm having some issues with the unfollowing part and the counter. I'll show a video demonstrating the issue.

**Comment by itsShaVawn** [comment]
[2022-03-12 22-22-04.zip](https://github.com/TradeShare490/TradeShare/files/8238917/2022-03-12.22-22-04.zip)

**Review by AlyaNaseer — APPROVED** [review]

**Review by AlyaNaseer — APPROVED** [review]

**Review by itsShaVawn — APPROVED** [review]
Nice work!

**Pull request merged**